presto-kinesis/src/test/java/io/prestosql/plugin/kinesis/util/MockKinesisClient.java

```diff
@@ -0,0 +1,604 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kinesis.util;
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.AmazonWebServiceRequest;
+import com.amazonaws.ResponseMetadata;
+import com.amazonaws.regions.Region;
+import com.amazonaws.services.kinesis.AmazonKinesisClient;
+import com.amazonaws.services.kinesis.model.CreateStreamRequest;
+import com.amazonaws.services.kinesis.model.CreateStreamResult;
+import com.amazonaws.services.kinesis.model.DescribeStreamRequest;
+import com.amazonaws.services.kinesis.model.DescribeStreamResult;
+import com.amazonaws.services.kinesis.model.GetRecordsRequest;
+import com.amazonaws.services.kinesis.model.GetRecordsResult;
+import com.amazonaws.services.kinesis.model.GetShardIteratorRequest;
+import com.amazonaws.services.kinesis.model.GetShardIteratorResult;
+import com.amazonaws.services.kinesis.model.ListStreamsRequest;
+import com.amazonaws.services.kinesis.model.ListStreamsResult;
+import com.amazonaws.services.kinesis.model.ListTagsForStreamRequest;
+import com.amazonaws.services.kinesis.model.ListTagsForStreamResult;
+import com.amazonaws.services.kinesis.model.PutRecordRequest;
+import com.amazonaws.services.kinesis.model.PutRecordResult;
+import com.amazonaws.services.kinesis.model.PutRecordsRequest;
+import com.amazonaws.services.kinesis.model.PutRecordsRequestEntry;
+import com.amazonaws.services.kinesis.model.PutRecordsResult;
+import com.amazonaws.services.kinesis.model.PutRecordsResultEntry;
+import com.amazonaws.services.kinesis.model.Record;
+import com.amazonaws.services.kinesis.model.SequenceNumberRange;
+import com.amazonaws.services.kinesis.model.Shard;
+import com.amazonaws.services.kinesis.model.StreamDescription;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Mock kinesis client for testing that is primarily used for reading from the
+ * stream as we do here in Presto.
+ * <p>
+ * This is to help prove that the API is being used correctly and debug any
+ * issues that arise without incurring AWS load and charges.  It is far from a complete
+ * implementation of Kinesis.
+ * <p>
+ */
+public class MockKinesisClient
+        extends AmazonKinesisClient
+{
+    private String endpoint = "";
+    private Region region;
+    List<InternalStream> streams = new ArrayList();
+
+    //// Support classes
+
+    public static class InternalShard
+            extends Shard
+    {
+        private List<Record> recs = new ArrayList<>();
+        private String streamName = "";
+        private int index;
+
+        public InternalShard(String owningStream, int anIndex)
```

Comment:
Name these parameters the same as the fields. This is allowed because the assignment disambiguates it:
```java
this.streamName = streamName;
this.index = index;

Reply:
changed